### PR TITLE
Add ui scale9 sprite

### PR DESCRIPTION
**Pull request by zilongshanren**
1. Add UIScale9Sprite  to mimic the function of extension::Scale9Sprite and enhance the Scale9Sprite class.
2. Make all the UI classes which used extension::Scale9Sprite now uses ui::Scale9Sprite instead.
3. add UIScale9Sprite Tests 
4. fix platform related compile errors

### CODE REVIEW

**Comment by zilongshanren** [comment]
[ci rebuild]

**Comment by zilongshanren** [comment]
[ci rebuild]

**Comment by darkdukey** [comment]
ImageView have scale9 too so which one are we going to use moving forward?

File: tests/cpp-tests/Classes/AppDelegate.cpp [hunk 1]
```diff
@@ -108,7 +108,7 @@ bool AppDelegate::applicationDidFinishLaunching()
     // a bug in DirectX 11 level9-x on the device prevents ResolutionPolicy::NO_BORDER from working correctly
     glview->setDesignResolutionSize(designSize.width, designSize.height, ResolutionPolicy::SHOW_ALL);
 #else
-    glview->setDesignResolutionSize(designSize.width, designSize.height, ResolutionPolicy::NO_BORDER);
+    glview->setDesignResolutionSize(designSize.width, designSize.height, ResolutionPolicy::SHOW_ALL);
```
Comment (dumganhar) [inline tests/cpp-tests/Classes/AppDelegate.cpp:111]: @zilongshanren, why to change resolution policy from NO_BORDER to SHOW_ALL?

Comment (zilongshanren) [reply tests/cpp-tests/Classes/AppDelegate.cpp:111]: @dumganhar 
it's a mistake...

Comment (dumganhar) [reply tests/cpp-tests/Classes/AppDelegate.cpp:111]: Correct this in https://github.com/cocos2d/cocos2d-x/pull/16996.

**Pull request merged**